transaction-status/src/parse_associated_token.rs

```diff
@@ -27,69 +27,56 @@ pub fn parse_associated_token(
-                        "source": account_keys[instruction.accounts[0] as usize].to_string(),
-                        "account": account_keys[instruction.accounts[1] as usize].to_string(),
-                        "wallet": account_keys[instruction.accounts[2] as usize].to_string(),
-                        "mint": account_keys[instruction.accounts[3] as usize].to_string(),
-                        "systemProgram": account_keys[instruction.accounts[4] as usize].to_string(),
-                        "tokenProgram": account_keys[instruction.accounts[5] as usize].to_string(),
-                    }),
-                })
-            }
-            AssociatedTokenAccountInstruction::CreateIdempotent => {
-                check_num_associated_token_accounts(&instruction.accounts, 6)?;
-                Ok(ParsedInstructionEnum {
-                    instruction_type: "createIdempotent".to_string(),
-                    info: json!({
-                        "source": account_keys[instruction.accounts[0] as usize].to_string(),
-                        "account": account_keys[instruction.accounts[1] as usize].to_string(),
-                        "wallet": account_keys[instruction.accounts[2] as usize].to_string(),
-                        "mint": account_keys[instruction.accounts[3] as usize].to_string(),
-                        "systemProgram": account_keys[instruction.accounts[4] as usize].to_string(),
-                        "tokenProgram": account_keys[instruction.accounts[5] as usize].to_string(),
-                    }),
-                })
-            }
-            AssociatedTokenAccountInstruction::RecoverNested => {
-                check_num_associated_token_accounts(&instruction.accounts, 7)?;
-                Ok(ParsedInstructionEnum {
-                    instruction_type: "recoverNested".to_string(),
-                    info: json!({
-                        "nestedSource": account_keys[instruction.accounts[0] as usize].to_string(),
-                        "nestedMint": account_keys[instruction.accounts[1] as usize].to_string(),
-                        "destination": account_keys[instruction.accounts[2] as usize].to_string(),
-                        "nestedOwner": account_keys[instruction.accounts[3] as usize].to_string(),
-                        "ownerMint": account_keys[instruction.accounts[4] as usize].to_string(),
-                        "wallet": account_keys[instruction.accounts[5] as usize].to_string(),
-                        "tokenProgram": account_keys[instruction.accounts[6] as usize].to_string(),
-                    }),
-                })
-            }
+        AssociatedTokenAccountInstruction::try_from_slice(&instruction.data)
+            .map_err(|_| ParseInstructionError::InstructionNotParsable(ParsableProgram::SplToken))?
+    };
+
+    match ata_instruction {
+        AssociatedTokenAccountInstruction::Create => {
+            check_num_associated_token_accounts(&instruction.accounts, 6)?;
```

Comment:
I think this couldn't parse the old ATA instruction which has 7 accounts 🤔

Reply:
Should be ok because `check_num_accounts` is only checking if `instruction.accounts.len()` isn't less than `6`. It doesn't care if the number of accounts is greater:

```rust
pub(crate) fn check_num_accounts(
    accounts: &[u8],
    num: usize,
    parsable_program: ParsableProgram,
) -> Result<(), ParseInstructionError> {
    if accounts.len() < num {
        Err(ParseInstructionError::InstructionKeyMismatch(
            parsable_program,
        ))
    } else {
        Ok(())
    }
}
```

Reply:
ah. sorry. I thought it is an exact match. 😞